ethicml/utility/data_structures.py

```diff
@@ -230,21 +234,33 @@ def info(self) -> Dict[str, float]:
     @staticmethod
     def from_npz(npz_path: Path) -> "Prediction":
         """Load prediction from npz file."""
+        info = None
+        if (npz_path.parent / "info.txt").exists():
+            with open(npz_path.parent / "info.txt") as json_file:
+                info = json.load(json_file)
         with npz_path.open("rb") as npz_file:
             data = np.load(npz_file)
             if "soft" in data:
-                return SoftPrediction(soft=pd.Series(np.squeeze(data["soft"])))
-            return Prediction(hard=pd.Series(np.squeeze(data["hard"])))
+                return SoftPrediction(soft=pd.Series(np.squeeze(data["soft"])), info=info)
+            return Prediction(hard=pd.Series(np.squeeze(data["hard"])), info=info)
 
     def to_npz(self, npz_path: Path) -> None:
         """Save prediction as npz file."""
+        if self.info:
+            for v in self.info.values():
+                assert isinstance(v, float), "Info must be Dict[str, float]"
+            json_path = npz_path.parent / 'info.txt'
```

Comment:
`.json` maybe? not that important though

```suggestion
            json_path = npz_path.parent / 'info.json'
```